lib/listen/record.rb

```diff
@@ -64,7 +64,10 @@ def _sub_tree(rel_path)
         if path == rel_path
           result.merge!(meta)
         else
-          sub_path         = path.sub(%r{\A#{rel_path}/?}, '')
+          parts = path.split("/")
+          next unless parts.include?(rel_path)
+
+          sub_path = (parts - [rel_path]).join("/")
```

Comment:
This will remove `rel_path` from anywhere in the path. Wouldn't that trip up with something like
```
a/b/c/cypress
```
(a file that happened to have that same name)
or
```
a/b/cypress/d
```
(a nested folder that happened to have that same name)

Reply:
And the `join` could use `File::SEPARATOR` too.

Reply:
I added another test for this case, but it seems to work fine.

Reply:
My understanding is this should only return files at `#{rel_path}/**/**`.

Reply:
Ah, you are correct. I was thinking `Array#-` worked as a set, but the docs are clear that it must match from the front, and that you need to a `Set` if you want `#-` to work like a set.

Reply:
Wait, I take that back.
```
irb> ['a', 'b', 'c'] - ['b']
=> ['a', 'c']

You only want to remove from the front, right?

Reply:
Oh I see what you mean. Yeah i suspect you’re right. Let me try your simpler looking code tomorrow.